### PR TITLE
Add 'fileDescriptors' option to allow file I/O to be used in the browser

### PR DESCRIPTION
**Pull request by kevinbarabash**
This option is only used by wasi-browser.ts.  It doesn't really make sense for node/deno/bun since those runtimes have native support for WASI.

I'm having issues running the tests so I haven't bothered trying to enable `fileAccess` for the browser tests.

I have done some manual testing to verify that things are working as expected.  I have a plugin with the following F# cord:
```fs
[<UnmanagedCallersOnly(EntryPoint = "greet")>]
let Greet () : int32 =
    printfn "Greet called"

    let contents = File.ReadAllText("./input.txt")
    printfn "contents = %s" contents
    
    let input = Pdk.GetInputString()
    printfn $"input = {input}"
    let data = JsonSerializer.Deserialize<Data> input
    
    printfn $"greeting = {data.greeting}, name = {data.name}"
    
    let result = $"{data.greeting}, {data.name}!"
    Pdk.SetOutput(result)
    0
```

The JavaScript host code looks like this (note, this requires input.txt to exist in the directory being served):
```js
import createPlugin from '@extism/extism';
import { WASI, File, OpenFile, ConsoleStdout, PreopenDirectory } from "@bjorn3/browser_wasi_shim";

const input = await fetch(`/input.txt`);
const inputBuffer = await input.arrayBuffer();

const url = new URL(`${location.protocol}/${location.host}/Plugin/bin/Debug/net8.0/wasi-wasm/AppBundle/Plugin.wasm`);
const options = { 
    useWasi: true, 
    enableWasiOutput: true,
    fileDescriptors: [
        new PreopenDirectory(".", {
            "input.txt": new File(inputBuffer),
        }),
    ]
};
const plugin = await createPlugin(url, options);

const input = JSON.stringify({ greeting: "hello", name: "world" });
const out = await plugin.call("greet", input);
console.log(out.text());
```

The output in the browser console looks like this:
```
Greet called
contents = This is a text file
input = {"greeting":"hello","name":"world"}
greeting = hello, name = world
hello, world!
```

### CODE REVIEW

File: src/polyfills/deno-wasi.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import { Fd } from '@bjorn3/browser_wasi_shim';
```
[the remark this answers is not in the document]
Comment (kevinbarabash) [reply src/polyfills/deno-wasi.ts:1]: Passing in an a `WASI` instance seems like the way to go.  I'll update this PR this weekend.

Do you know what's up with the tests?  I tried running the tests on CI without any code changes in https://github.com/escalier-lang/js-sdk/pull/2 and am seeing the same failure in that PR as in this PR.

[the remark this answers is not in the document]
Comment (chrisdickinson) [reply src/polyfills/deno-wasi.ts:1]: Ah, yeah – there was [a `node:worker_threads` regression](https://github.com/denoland/deno/issues/23070) on upstream Deno; it sounds like they've got a fix for it, though.

[the remark this answers is not in the document]
Comment (kevinbarabash) [reply src/polyfills/deno-wasi.ts:1]: I didn't get to it this weekend.  Next weekend is a long weekend so I'll have more time.

**Comment by chrisdickinson** [comment]
Going to close this one for now – we're going to revisit the WASI integration for Deno 2; in the process we'll also try to make sure that it's possible to provide overrides for `fileDescriptors` in-browser.

**Pull request closed without merging**